pulp_container/app/tasks/sync_stages.py

```diff
@@ -443,12 +444,15 @@ def _get_content_data_blocking():
                 dl_res.artifact_attributes
             )
             content_data = json.loads(raw_data)
+            media_type = manifest_data.get("mediaType") or dl_res.headers.get("content-type")
+            if media_type is None:
+                raise ManifestInvalid(dl_res.artifact_attributes["sha256"])
             manifest = Manifest(
                 digest=manifest_data["digest"],
                 schema_version=2
-                if manifest_data["mediaType"] in (MEDIA_TYPE.MANIFEST_V2, MEDIA_TYPE.MANIFEST_OCI)
+                if media_type in (MEDIA_TYPE.MANIFEST_V2, MEDIA_TYPE.MANIFEST_OCI)
                 else 1,
-                media_type=manifest_data["mediaType"],
```

Comment:
`manifest_data` is a piece of information extracted from the manifest_list about a single image manifest which we are downloading and saving here. Based on the specs, media_type is a required field so it will be present.